### DISCUSSION TITLE
Current Docker image for JDK?

### DISCUSSION THREAD

**Discussion by StefanOltmann** [libvips/libvips#4318] [Q&A] [answered]
I want to use libvips in a Docker container with a Java web service.

I found this older sample:
https://github.com/jcupitt/docker-builds/blob/master/libvips-openjdk-alpine/Dockerfile

What I changed:
- `FROM amazoncorretto:22-alpine`
- `VIPS_VERSION=8.16.0`
- The suffix `gz` to `xz`

It fails with `/bin/sh: ./configure: not found`

Is that a bad Docker base image or what's the problem here?

**Comment by jcupitt** [comment]
Hello @StefanOltmann,

libvips uses meson and ninja to build now, so you need to install them. You can adapt one of the more recent dockerfiles there, perhaps:

https://github.com/jcupitt/docker-builds/blob/master/libvips-ubuntu22.04/Dockerfile

**Reply by StefanOltmann** [reply]
`amazoncorretto:22-alpine` does not seem to have `apt-get`.
So I will need to try another Dockerfile.
They're all pretty old unfortunately.

**Reply by StefanOltmann** [reply]
https://github.com/jcupitt/docker-builds/tree/48ed9cc24c0245adbedc59f96433701f0e978c86/libvips-amazonlinux2 seems to be the only one with amazoncorretto in it, but it's not alpine and pretty big.

Also, it failed in build for me with the current versions.

I don't know if there are other JDK samples...

Also, I have the question if there is a sample for a multi-stage build.

I use that because it's advertised as yielding in smaller images.

Shouldn't it be possible to build vips in BUILD_STAGE and copy it over to the runtime image?

```
FROM amazoncorretto:22 AS BUILD_STAGE
WORKDIR /tmp
#COPY .git .git
COPY gradle gradle
COPY build.gradle.kts gradle.properties settings.gradle.kts gradlew ./
COPY src src
RUN ./gradlew --no-daemon --info buildFatJar

FROM amazoncorretto:22-alpine

# TODO Add vips

RUN mkdir /app
COPY --from=BUILD_STAGE /tmp/build/libs/*-all.jar /app/ktor-server.jar

EXPOSE 8080:8080

ENTRYPOINT ["java","-Xlog:gc+init","-XX:+PrintCommandLineFlags","-jar","/app/ktor-server.jar"]

```

**Reply by jcupitt** [reply]
Do you want to use an alpine base? Alpine comes with libvips, you don't need to build from source. You can get a recent, pre-built libvips for most linuxes, it really depends on your needs.

**Reply by StefanOltmann** [reply]
At least vips seems not to be pre-installed in `amazoncorretto:22-alpine`.

**Reply by StefanOltmann** [reply]
How would I install it?

`RUN apk add --no-cache vips` runs, but `RUN vips -l` after that (for test purposes) fails.

**Reply by jcupitt** [reply, marked as answer]
The alpine `vips` package just has the library. You need `vips-tools` for the CLI interface.

**Reply by StefanOltmann** [reply]
Ah, I see. Thank you.
